### ISSUE TITLE
Make modules documentation more prominent and add simple examples

### ISSUE THREAD

**Issue by ataraxer** [circe/circe#704]
Circe is a superb library which provides a lot of useful functionality, but it's spread across a lot of modules and it's not always obvious where to look for a particular thing.

This issue aims to address it by providing an overview of modules in [documentation](https://circe.github.io/circe/) with simple examples of usage. It might also be good to accompany each module's description with a short motivation note to explain why the decision to separate some functionality was made.

**Comment by ataraxer** [comment]
I'm going to start by going through the list of modules at http://circe.github.io/circe/api/ to see if I can figure out how to present each one of them. Some I've already used and have an idea of what might be the most important things to demonstrate to get people started faster, others I did not, so I'll be able to note what might be the most confusing things about using them (if any) from a "clean" perspective.

Hopefully I'll have some time to start drafting a PR this weekend.
